src/app/core/auth/auth.interceptor.ts

```diff
@@ -49,17 +52,65 @@ export class AuthInterceptor implements HttpInterceptor {
   }
 
   private isLoginResponse(http: HttpRequest<any> | HttpResponseBase): boolean {
-    return http.url && http.url.endsWith('/authn/login');
+    return http.url && http.url.endsWith('/authn/login')
   }
 
   private isLogoutResponse(http: HttpRequest<any> | HttpResponseBase): boolean {
     return http.url && http.url.endsWith('/authn/logout');
   }
 
-  private makeAuthStatusObject(authenticated:boolean, accessToken?: string, error?: string): AuthStatus {
+  private parseShibbolethLocation(unparsedLocation: string): string {
+    let parsedLocation = '';
+    unparsedLocation = unparsedLocation.trim();
+    unparsedLocation = unparsedLocation.replace('location="', '');
+    unparsedLocation = unparsedLocation.replace('"', '');
+    let re = /%3A%2F%2F/g;
+    unparsedLocation = unparsedLocation.replace(re, '://');
+    re = /%3A/g
+    unparsedLocation = unparsedLocation.replace(re, ':')
+    parsedLocation = unparsedLocation + '/shibboleth';
+
+    return parsedLocation;
+  }
+
+  private parseAuthMethodsfromHeaders(headers: HttpHeaders): AuthMethodModel[] {
+    const authMethodModels: AuthMethodModel[] = [];
+    const parts: string[] = headers.get('www-authenticate').split(',');
+    // get the login methods names
+    // tslint:disable-next-line:forin
+    for (const i in parts) {
+      const part: string = parts[i].trim();
+      if (part.includes('realm')) {
+        const methodName = part.split(' ')[0];
+        const authMethod: AuthMethodModel = new AuthMethodModel(methodName);
+        // check if the authentication method is  shibboleth
+        // if so the next part is the shibboleth location
+        // e.g part i: shibboleth realm="DSpace REST API", part i+1:  location="/Shibboleth.sso/Login?target=https://serverUrl"
+        if (methodName.includes('shibboleth')) {
```

Comment:
is it possible to have a generic function to retrieve authentication method? this should avoid to manually add other check like this for every new authentication method that we would like to implement in the future
I think [this is](https://github.com/DSpace/DSpace/blob/d896fed7a71da0b4ad123ca541367dac9436ccb0/dspace-server-webapp/src/main/webapp/login.html#L103) a good example on how to implement it

Reply:
Tryed to make it more generic....please check it.